proxy/src/main/java/net/md_5/bungee/connection/DownstreamBridge.java

```diff
@@ -743,13 +745,32 @@ public void handle(Respawn respawn)
     @Override
     public void handle(Commands commands) throws Exception
     {
+        final Collection<Map.Entry<String, Command>> unmodifiedCommandMap = bungee.getPluginManager().getCommands();
+        final ProxyDefineCommandsEvent event = new ProxyDefineCommandsEvent( server, con, unmodifiedCommandMap );
```

Comment:
I think there is a design question about whether the event includes all commands or just bungee commands. Including all commands would give more flexibility (like the Spigot event https://hub.spigotmc.org/javadocs/spigot/org/bukkit/event/player/PlayerCommandSendEvent.html)

Reply:
Ahh you're right, I forgot that it only has the bungee commands. The other two places that you left comments are because of that.

Supporting non-bungee commands doesn't add extra features if they can edit downstream commands on the downstream, but it would be convenient to have it in one place.

As it stands the event handler gets complete information with the `Command` type. To support non-bungee commands we would have do one of the following:
1. Limit the information they get to support both sources (maybe even just a key).
2. Create a new union type of Command and another new type or the brigadier node.

I don't like either option just to get plugin convenience, so unless you can think of another then I'm in favour of committing as is but with the removal of the additional assignments of `modified`.